tests/ashley/api/test_api_manage_moderators.py

```diff
@@ -211,13 +212,184 @@ def test_access_can_manage_moderators_list_non_moderators(self):
+        assign_perm("can_manage_moderator", user, forum, True)
+        # Creates the session
+        self.client.force_login(user, "ashley.auth.backend.LTIBackend")
+        session = self.client.session
+        session[SESSION_LTI_CONTEXT_ID] = lti_context.id
+        session.save()
+        response = self.client.post(
+            f"/api/v1.0/users/{user.id}/",
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 405)
+        self.assertEqual(response.json(), {"detail": 'Method "POST" not allowed.'})
+
+    def test_access_api_update_instructor_patch(self):
+        """
+        Standard instructor call to patch user is allowed but doesn't change anything,
+        as serializer's attributes are read-only.
+        """
+        # Creates a forum
+        user = UserFactory(id=1, public_username="user")
+        lti_context = LTIContextFactory(lti_consumer=user.lti_consumer)
+        forum = ForumFactory()
+        forum.lti_contexts.add(lti_context)
+        lti_context.sync_user_groups(user, ["instructor"])
+        assign_perm("can_manage_moderator", user, forum, True)
+        # Creates the session
+        self.client.force_login(user, "ashley.auth.backend.LTIBackend")
+        session = self.client.session
+        session[SESSION_LTI_CONTEXT_ID] = lti_context.id
+        session.save()
+        data = {"id": 10, "public_username": "Sab"}
+        response = self.client.patch(
+            f"/api/v1.0/users/{user.id}/",
+            json.dumps(data),
+            content_type="application/json",
+        )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), {"id": 1, "public_username": "user"})
+
+    def test_access_api_update_instructor_put(self):
+        """Standard instructor call to put user is allowed but doesn't change anything."""
+        # Creates a forum
+        user = UserFactory(id=1, public_username="user")
+        lti_context = LTIContextFactory(lti_consumer=user.lti_consumer)
+        forum = ForumFactory()
+        forum.lti_contexts.add(lti_context)
+        lti_context.sync_user_groups(user, ["instructor"])
+        assign_perm("can_manage_moderator", user, forum, True)
+        # Creates the session
+        self.client.force_login(user, "ashley.auth.backend.LTIBackend")
+        session = self.client.session
+        session[SESSION_LTI_CONTEXT_ID] = lti_context.id
+        session.save()
+        data = {"id": 10, "public_username": "Sab"}
+        response = self.client.put(
+            f"/api/v1.0/users/{user.id}/",
+            json.dumps(data),
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), {"id": 1, "public_username": "user"})
+
+    def test_access_api_moderator_group_anonymous_patch(self):
+        """anonymous can't patch user to add or remove the group moderator."""
```

Comment:
not patching user anymore
```suggestion
        """anonymous user is not allowed to add or remove a user's moderator group via the action endpoint."""
```